src/main.c

```diff
@@ -51,14 +51,19 @@ static int load_hd()
 	int ret = 0;
 	const char *disk1 = fbc_get_string("hard_disk", "disk1");
 	const char *disk2 = fbc_get_string("hard_disk", "disk2");
+	int sectors_per_track = fbc_get_int("hard_disk", "sectors_per_track");
+	int cylinders = fbc_get_int("hard_disk", "cylinders");
+	// bytes per sector is fixed at 512, not configurable, all hard drives of the 3B1
+	// era used 512-byte sectors.
+	const int bytes_per_sector = 512;
 
 	state.hdc_disc0 = fopen(disk1, "r+b");
 	if (!state.hdc_disc0){
 		fprintf(stderr, "Drive 0: ERROR loading disc image '%s'.\n", disk1);
 		state.hdc_disc0 = NULL;
 		return (0);
 	} else {
-		if (wd2010_init(&state.hdc_ctx, state.hdc_disc0, 0, 512, 17, 8) == WD2010_ERR_OK) {
+		if (wd2010_init(&state.hdc_ctx, state.hdc_disc0, 0, bytes_per_sector, sectors_per_track, cylinders) == WD2010_ERR_OK) {
```

Comment:
The last parameter is "heads" not cylinders -- cylinders is calculated based on hd.img filesize / (heads * sectors_per_track * bytes_per_sector)